dev/io.openliberty.microprofile.internal_fat/fat/src/io/openliberty/microprofile/internal/test/suite/MPCompatibleTest.java

```diff
@@ -0,0 +1,346 @@
+                                          false, //cleanStart
+                                          false, //validateApps
+                                          false, //expectStartFailure
+                                          false); //validateTimedExit
+        } finally {
+            server.stopServer("CWWKF0033E: The singleton features microProfile-3.3 and microProfile-4.0 cannot be loaded at the same time.",
+                              "CWWKF0046W: The configuration includes an incompatible combination of features");
+        }
+    }
+
+    /**
+     * microProfile-4.0 plus mpConfig-1.4
+     * Should fail because microProfile-4.0 can not be started at the same time as mpConfig-1.4
+     *
+     * @throws Exception
+     */
+    @Test
+    @Mode(TestMode.FULL)
+    @ExpectedFFDC("java.lang.IllegalArgumentException")
+    public void testMP40andMPConfig14() throws Exception {
+        try {
+            server.setServerConfigurationFromFilePath("MP40andMPConfig14.xml");
+            server.startServerAndValidate(true, //preClean
+                                          false, //cleanStart
+                                          false, //validateApps
+                                          false, //expectStartFailure
+                                          false); //validateTimedExit
+        } finally {
+            server.stopServer("CWWKF0033E: The singleton features mpConfig-1.4 and mpConfig-2.0 cannot be loaded at the same time.",
+                              "CWWKF0046W: The configuration includes an incompatible combination of features");
+        }
+    }
+
+    /**
+     * microProfile-4.0 plus mpContextPropagation-1.0
+     * Should fail because microProfile-4.0 can not be started at the same time as mpContextPropagation-1.0
+     *
+     * @throws Exception
+     */
+    @Test
+    @Mode(TestMode.LITE)
+    @ExpectedFFDC("java.lang.IllegalArgumentException")
+    public void testMP40andCtxPropagtion10() throws Exception {
+        try {
+            server.setServerConfigurationFromFilePath("MP40andMPCtx10.xml");
+            server.startServerAndValidate(true, //preClean
+                                          false, //cleanStart
+                                          false, //validateApps
+                                          false, //expectStartFailure
+                                          false); //validateTimedExit
+        } finally {
+            server.stopServer("CWWKF0033E: The singleton features io.openliberty.mpCompatible-0.0 and io.openliberty.mpCompatible-4.0 cannot be loaded at the same time.",
+                              "CWWKF0046W: The configuration includes an incompatible combination of features");
+        }
+    }
+
+    /**
+     * microProfile-4.0 plus mpContextPropagation-1.1
+     * Should always pass
+     *
+     * @throws Exception
+     */
+    @Test
+    @Mode(TestMode.LITE)
+    public void testMP40andCtxPropagtion11() throws Exception {
```

Comment:
Context Propagation won't be released when MP 4.0 is out. Will this cause any issue?

Reply:
No, these tests even run against no-ship features